### PR TITLE
Add filter for status on listings

### PR DESCRIPTION
**Pull request by pmargreff**
Add status filter as requested by @gabiseabra. 

Using an array to allow multiple statuses, as our next steps will probably add more options in this field, and using arrays, we wouldn't break any public API... 

Also testing corner cases on filter module test and test integration (GraphQL request) in a different test block.

### CODE REVIEW

File: apps/re_web/lib/graphql/types/listing.ex [hunk 1]
```diff
@@ -203,6 +203,7 @@ defmodule ReWeb.Types.Listing do
     field :garage_types, list_of(:garage_type)
     field :cities, list_of(:string)
     field :cities_slug, list_of(:string)
+    field :statuses, list_of(:string)
```
Comment (rhnonose) [inline apps/re_web/lib/graphql/types/listing.ex]: This should be `list_of(non_null(:string))`

**Comment by pmargreff** [comment]
testing coveralls

File: apps/re_web/lib/graphql/types/listing.ex (hunk 1 above)
Comment (pmargreff) [reply apps/re_web/lib/graphql/types/listing.ex]: So, shouldn't be other filters as well? Should we update them?

Comment (rhnonose) [reply apps/re_web/lib/graphql/types/listing.ex]: The trouble with updating other fields is that it might break the queries that defines the types as list of "might be null", so might require syncing with the clients.
